components/post_markdown/index.test.ts

```diff
@@ -63,16 +69,16 @@ describe('makeGetMentionKeysForPost', () => {
                 myPreferences: {},
             },
         },
-    };
+    } as unknown as GlobalState;
```

Comment:
A more type safe method instead of using "as unknown as GlobalState" is to force type the individual keys instead. eg.
```
entities: {
            users: {
                currentUserId: user.id,
                profiles: {
                    [user.id]: user,
                },
            } as UsersState
```

Reply:
@M-ZubairAhmed, first of all, this doesn’t work as typescript screams for missing properties. Second, if I do this for the `userState` then I need to do it for others as well.
<img width="1011" alt="Screenshot 2022-10-01 at 3 52 21 PM" src="https://user-images.githubusercontent.com/96474572/193405012-0bd92518-d815-4b30-8d76-416ddd923cb5.png">
<img width="425" alt="Screenshot 2022-10-01 at 3 52 50 PM" src="https://user-images.githubusercontent.com/96474572/193405008-af901d67-ee82-46a7-99af-d546dc81204f.png">

I don’t have any issue with this, but it would be great if you could point to an example where all the required properties of the Global state are present. (same goes for other suggestions related to this).

Reply:
This was throwing error not because its missing all the properties of GlobalState, but because of `enitities.groups.members` is not a valid property in the store.
Without conversion to unknown the following works
```ts
    const baseState = {
        entities: {
            users: {
                currentUserId: user.id,
                profiles: {
                    [user.id]: user,
                },
            },
            groups: {
                syncables: {},
                // members: {}, this is not a valid key
                groups: {
                    [group.id]: group,
                },
                myGroups: [
                    group.id,
                ],
            },
            teams: {
                teams: {
                    [team.id]: team,
                },
                groupsAssociatedToTeam: {
                    [team.id]: {ids: []},
                },
            },
            channels: {
                channels: {
                    [channel.id]: channel,
                },
                groupsAssociatedToChannel: {
                    [channel.id]: {ids: [group]},
                },
            },
            general: {
                config: {},
            },
            preferences: {
                myPreferences: {},
            },
        },
    } as GlobalState;
```